### PR TITLE
docs: add pnpm docs

### PR DESCRIPTION
**Pull request by Mister-Hope**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -28,7 +28,25 @@ footer: MIT Licensed | Copyright © 2018-present Evan You
 ### As Easy as 1, 2, 3
 
 <CodeGroup>
-  <CodeGroupItem title="YARN" active>
+  <CodeGroupItem title="PNPM" active>
+
+```bash
+# install in your project
+pnpm add -D vue @vuepress/client@nextvuepress@next
```
Comment (meteorlxy) [inline docs/README.md:35]: - put vuepress first
- missing whitespace

File: docs/guide/deployment.md [hunk 2]
```diff
@@ -43,20 +43,20 @@ jobs:
...
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           # choose node.js version to use
           node-version: '14'
 
       # cache node_modules
       - name: Cache dependencies
-        uses: actions/cache@v2
+        uses: actions/cache@v3
```
Comment (meteorlxy) [inline docs/guide/deployment.md:59]: Remove cache action, and use `cache` option of setup-node action

File: docs/guide/getting-started.md [hunk 3]
```diff
@@ -46,7 +55,15 @@ npm init
 - **Step 3**: Install VuePress locally
 
 <CodeGroup>
-  <CodeGroupItem title="YARN" active>
+  <CodeGroupItem title="PNPM" active>
+
+```bash
+pnpm add -D vue @vuepress/client@next vuepress@next
```
Comment (meteorlxy) [inline docs/guide/getting-started.md:61]: Same here

File: docs/guide/deployment.md [hunk 4]
```diff
@@ -87,6 +79,7 @@ jobs:
           # @see https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 ```
+
```
Comment (meteorlxy) [inline docs/guide/deployment.md:82]: Avoid this kind of formatting (and below)

**Review by meteorlxy — CHANGES_REQUESTED** [review]

File: docs/zh/guide/deployment.md [hunk 5]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+      - uses: actions/checkout@v3
         with:
           # “最近更新时间” 等 git 日志相关信息，需要拉取全部提交记录
           fetch-depth: 0
 
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          # 选择要使用的 pnpm 版本
+          version: 7
```
Comment (Mister-Hope) [inline docs/zh/guide/deployment.md:59]:
```suggestion
          version: 8
```

File: docs/guide/deployment.md [hunk 6]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+      - uses: actions/checkout@v3
         with:
           # fetch all commits to get last updated time or other git log info
           fetch-depth: 0
 
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          # choose pnpm version to use
+          version: 7
```
Comment (Mister-Hope) [inline docs/guide/deployment.md:59]:
```suggestion
          version: 8
```

**Comment by azat-io** [comment]
Might be worth merge it?

File: docs/guide/deployment.md [hunk 7]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+          # install deps with pnpm
+          run_install: true
+
       - name: Setup Node.js
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           # choose node.js version to use
-          node-version: '14'
+          node-version: 16
```
Comment (Mister-Hope) [inline docs/guide/deployment.md:67]:
```suggestion
          node-version: 18
```

File: docs/guide/deployment.md [hunk 8]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+      - uses: actions/checkout@v3
         with:
           # fetch all commits to get last updated time or other git log info
           fetch-depth: 0
 
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          # choose pnpm version to use
+          version: 8
```
Comment (Mister-Hope) [inline docs/guide/deployment.md:59]:
```suggestion
          version: 7
```

File: docs/guide/deployment.md [hunk 9]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+      - uses: actions/checkout@v3
+        with:
+          # fetch all commits to get last updated time or other git log info
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          # choose node.js version to use
+          node-version: '14'
```
Comment (Mister-Hope) [inline docs/guide/deployment.md:117]:
```suggestion
          node-version: 18
```

File: docs/zh/guide/deployment.md [hunk 10]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+          # 使用 pnpm 安装依赖
+          run_install: true
+
       - name: Setup Node.js
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           # 选择要使用的 node 版本
-          node-version: '14'
+          node-version: 16
```
Comment (Mister-Hope) [inline docs/zh/guide/deployment.md:67]:
```suggestion
          node-version: 18
```

File: docs/zh/guide/deployment.md [hunk 11]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+      - uses: actions/checkout@v3
         with:
           # “最近更新时间” 等 git 日志相关信息，需要拉取全部提交记录
           fetch-depth: 0
 
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          # 选择要使用的 pnpm 版本
+          version: 8
```
Comment (Mister-Hope) [inline docs/zh/guide/deployment.md:59]:
```suggestion
          version: 7
```

File: docs/zh/guide/deployment.md [hunk 12]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+      - uses: actions/checkout@v3
+        with:
+          # “最近更新时间” 等 git 日志相关信息，需要拉取全部提交记录
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          # 选择要使用的 node 版本
+          node-version: '14'
```
Comment (Mister-Hope) [inline docs/zh/guide/deployment.md:117]:
```suggestion
          node-version: 18
```

**Comment by Mister-Hope** [comment]
> Might be worth merge it?
> 
> **upd.** I think the order should be: npm, pnpm, yarn. This seems more logical.

pnpm takes less time during install comparing to npm. so we do recommand people to use pnpm as first choise.

**Comment by meteorlxy** [comment]
We may also need to update the deployment guide to use pnpm as recommendation

**Comment by Mister-Hope** [comment]
Done @meteorlxy

File: docs/guide/deployment.md [hunk 13]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+          # deploy the default output dir of VuePress
+          build_dir: docs/.vuepress/dist
+        env:
+          # @see https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+```
+
+  </CodeGroupItem>
+
+  <CodeGroupItem title="YARN">
```
Comment (meteorlxy) [inline docs/guide/deployment.md:91]: We could remove the yarn tab then, as we are assuming user is using pnpm

File: docs/guide/deployment.md [hunk 14]
```diff
@@ -198,21 +263,21 @@ See [Layer0 Documentation > Framework Guides > VuePress](https://docs.layer0.co/
...
 ## Vercel
 
 1. Go to [Vercel](https://vercel.com), set up a new project from GitHub with the following settings:
 
-    - **FRAMEWORK PRESET:** `Other`
-    - **BUILD COMMAND:** `yarn docs:build` 
-    - **OUTPUT DIRECTORY:** `docs/.vuepress/dist`
-   
+   - **FRAMEWORK PRESET:** `Other`
+   - **BUILD COMMAND:** `yarn docs:build`
```
Comment (meteorlxy) [inline docs/guide/deployment.md:280]: pnpm

File: docs/zh/guide/deployment.md [hunk 15]
```diff
@@ -43,27 +47,76 @@ jobs:
...
+          # 部署目录为 VuePress 的默认输出目录
+          build_dir: docs/.vuepress/dist
+        env:
+          # @see https://docs.github.com/cn/actions/reference/authentication-in-a-workflow#about-the-github_token-secret
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+```
+
+  </CodeGroupItem>
+
+  <CodeGroupItem title="YARN">
```
Comment (meteorlxy) [inline docs/zh/guide/deployment.md:91]: Same here

File: docs/zh/guide/deployment.md [hunk 16]
```diff
@@ -199,22 +263,22 @@ heroku login
 
 1. 前往 [Netlify](https://netlify.com) ，从 GitHub 创建一个新项目，并进行如下配置：
 
-    - **Build Command:** `yarn docs:build`
-    - **Publish directory:** `docs/.vuepress/dist`
+   - **Build Command:** `yarn docs:build`
```
Comment (meteorlxy) [inline docs/zh/guide/deployment.md:266]: Same here

File: docs/guide/deployment.md [hunk 17]
```diff
@@ -198,21 +200,21 @@ See [Layer0 Documentation > Framework Guides > VuePress](https://docs.layer0.co/
 
 1. On [Netlify](https://netlify.com), set up a new project from GitHub with the following settings:
 
-    - **Build Command:** `yarn docs:build`
-    - **Publish directory:** `docs/.vuepress/dist`
+   - **Build Command:** `npm i -g pnpm@7 && pnpm docs:build`
```
Comment (meteorlxy) [inline docs/guide/deployment.md:203]: Is it necessary to add `@7` for pnpm? If not, we'd better remove it so we don't need to update it in the future

File: docs/zh/guide/deployment.md [hunk 18]
```diff
@@ -161,7 +162,7 @@ pages:
 }
 ```
 
-3. 在执行了 `yarn docs:build` 或 `npm run docs:build` 后, 使用 `firebase deploy` 指令来部署。
+3. 在执行了 `pnpm docs:build`, `yarn docs:build` 或 `npm run docs:build` 后, 使用 `firebase deploy` 指令来部署。
```
Comment (meteorlxy) [inline docs/zh/guide/deployment.md:165]: Same here

File: docs/zh/guide/deployment.md [hunk 19]
```diff
@@ -199,22 +200,22 @@ heroku login
 
 1. 前往 [Netlify](https://netlify.com) ，从 GitHub 创建一个新项目，并进行如下配置：
 
-    - **Build Command:** `yarn docs:build`
-    - **Publish directory:** `docs/.vuepress/dist`
+   - **Build Command:** `npm i -g pnpm@7 && pnpm docs:build`
```
Comment (meteorlxy) [inline docs/zh/guide/deployment.md:203]: Same here

**Review by meteorlxy — CHANGES_REQUESTED** [review]

**Pull request merged**